pkg/scheduler/scheduler.go

```diff
@@ -57,23 +57,6 @@ const (
 	pluginMetricsSamplePercent = 10
 )
 
-// podConditionUpdater updates the condition of a pod based on the passed
```

Comment:
I'm kind of having second thoughts about this.

Maybe we should still get rid of the interfaces (and fakes), but preserve the implementation in a (single) separate struct called podClient.

Thoughts @ahg-g ?